### PR TITLE
Remove dependency on lodash.isplainobject

### PR DESCRIPTION
**Pull request by BlakeStearman**
### Changes
* feat: Remove dependency on lodash._isPlainObject.

  Adds local implementation of `isPlainObject` as well as unit tests.

### CODE REVIEW

File: src/plainObject.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"use strict";
```
Comment (dasa) [inline src/plainObject.ts:1]: Remove

**Review by dasa — CHANGES_REQUESTED** [review]

**Comment by dasa** [comment]
Can the Node 10 and 12 tests be removed since those versions are no longer [maintained](https://github.com/nodejs/Release/#end-of-life-releases)?

File: src/plainObject.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+// import ldIsPlainObject from "lodash.isplainobject";
```
Comment (dasa) [inline src/plainObject.test.ts:1]: Remove line

**Review by dasa — APPROVED** [review]

**Comment by dasa** [comment]
It'd be good to add Node 18 to the tests.

**Review by dasa — APPROVED** [review]

**Review by dasa — APPROVED** [review]
👍🏻

**Review by gbrown-gis — APPROVED** [review]
Verified that library is removed and replaced with our own file implementation.

**Review by ssylvia — APPROVED** [review]

**Pull request merged**